### PR TITLE
Return device index

### PR DESCRIPTION
**Pull request by planetA**
Add wrapper for ibv_get_device_index.

### CODE REVIEW

**Comment by planetA** [comment]
This commit adds an implicit dependency on Rust 1.53, because of `ErrorKind::Unsupported`.

File: src/lib.rs [hunk 1]
```diff
@@ -259,6 +259,19 @@ impl<'devlist> Device<'devlist> {
...
     }
+
+    /// Returns stable IB device index as it is assigned by the kernel
+    /// # Errors
+    ///
+    ///  - `ENOTSUP`: Stable index is not supported
+    pub fn index(&self) -> io::Result<i32> {
+        let idx = unsafe { ffi::ibv_get_device_index(*self.0) };
+        if idx == -1 {
+            Err(io::Error::new(io::ErrorKind::Unsupported, "device index not known"))
```
[the remark this answers is not in the document]
Comment (planetA) [reply src/lib.rs:270]:
```
int ibv_get_device_index(struct ibv_device *device)
{
        struct verbs_sysfs_dev *sysfs_dev = verbs_get_device(device)->sysfs;

        return sysfs_dev->ibdev_idx;
}
```

This is the full source of the function. It does not set `errno`. The man page says:

> ibv_get_device_index()  returns  an  index,  or  -1  if  the kernel doesn’t support device
>  indexes.

File: src/lib.rs [hunk 2]
```diff
@@ -259,6 +259,19 @@ impl<'devlist> Device<'devlist> {
...
+        } else {
+            Ok(idx)
```
[the remark this answers is not in the document]
Comment (planetA) [reply src/lib.rs:272]: Yes, it is host-endian. It is an integer that is read from netlink as `u32`

File: src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonhoo) [reply src/lib.rs:270]: Great, thanks for checking!

**Review by jonhoo — APPROVED** [review]
Thanks!

**Pull request merged**